actions/docker-publish/action.yaml

```diff
@@ -1,60 +1,61 @@
+    required: true
+  docker-registry:
+    description: "Host where the image should be pushed to."
+    required: true
+  github-token:
+    description: "Github token."
+    required: true
+  ref:
+    description: "Ref to checkout"
     required: false
-    default: "."
+    default: ${{ github.event.repository.default_branch }}
 runs:
   using: "composite"
   steps:
-    - name: Download Docker image tar artifact
-      uses: actions/download-artifact@v3
-      with:
-        name: ${{ inputs.image-artifact-name }}
-        path: build
-
-    - name: Login to DockerHub
-      uses: docker/login-action@v1
+    - name: Check out repository
+      uses: actions/checkout@v3
       with:
-        username: ${{ inputs.username }}
-        password: ${{ inputs.password }}
+        ref: ${{ inputs.ref }}
+        token: ${{ inputs.github-token }}
 
-    - name: Publish Docker image
-      run: |
-        if [[ $(ls -1 build/*.tar 2>/dev/null | wc -l) != 1 ]]; then
-          >&2 echo "Error: A single image tar file is needed in the downloaded artifact. You can upload one before using this action: https://github.com/actions/upload-artifact."
-          exit 1
-        fi
+    # setup-buildx action will create and boot a builder using by default the docker-container driver.
+    # This is not required but recommended using it to be able to build multi-platform images, export cache, etc.
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2.2.1
 
-        docker load --input build/*.tar
+    - name: Login to the Registry
+      uses: "docker/login-action@v2.1.0"
+      with:
+        registry: "${{ inputs.docker-registry }}"
+        username: "${{ inputs.username }}"
+        password: "${{ inputs.password }}"
 
-        if [[ "$GITHUB_REF" =~ ^refs/tags/.* ]]; then
-          docker tag ${{ inputs.image-name }}:${{ github.run_id }} ${{ inputs.publisher }}/${{ inputs.image-name }}:latest
-          docker tag ${{ inputs.image-name }}:${{ github.run_id }} ${{ inputs.publisher }}/${{ inputs.image-name }}:${GITHUB_REF/refs\/tags\//}
-          docker push ${{ inputs.publisher }}/${{ inputs.image-name }}:latest
-          docker push ${{ inputs.publisher }}/${{ inputs.image-name }}:${GITHUB_REF/refs\/tags\//}
-        else
-          docker tag ${{ inputs.image-name }}:${{ github.run_id }} ${{ inputs.publisher }}/${{ inputs.image-name }}:${{ github.run_id }}
-          docker push ${{ inputs.publisher }}/${{ inputs.image-name }}:${{ github.run_id }}
-        fi
-      shell: bash
-      working-directory: ${{ inputs.working-directory }}
+    - name: Build and push
+      uses: docker/build-push-action@v3.2.0
+      with:
+        context: "${{ inputs.working-directory }}"
+        push: true
+        tags: |
+          ${{ inputs.docker-registry }}/${{ inputs.publisher }}/${{ inputs.image-name }}:${{ inputs.image-tag }}
+          ${{ inputs.docker-registry }}/${{ inputs.publisher }}/${{ inputs.image-name }}:latest
```

Comment:
Why did you change the image tag logic? This is a big breaking change and shouldn't be done unless it is specifically required. Especially we wouldn't want to publish `latest` accidentally for example

Reply:
i am using it in [another CI](https://github.com/bakdata/cluster-shell/pull/1) where i release a new tag and then use the new tag to build and publish a new docker image and also update the index.yaml on `gh-pages`.

Reply:
How do you support pushing dev images from feature or fix branches like this? Developers need to be able to test there image before merging without overwriting the latest tag